File: #1.data-structures/#2.Stack/src/java/src/modules/Stack.java
import java.util.ArrayList;
import java.util.List;

public class Stack<T> {
    private final List<T> items = new ArrayList<>();

    public void push(T item) {
        items.add(item);
    }

    public T pop() {
        if (items.isEmpty()) {
            return null;
        }
        return items.remove(items.size() - 1);
    }

    public T peek() {
        if (items.isEmpty()) {
            return null;
        }
        return items.get(items.size() - 1);
    }

    public boolean isEmpty() {
        return items.isEmpty();
    }

    public int size() {
        return items.size();
    }

    public void clear() {
        items.clear();
    }

    public List<T> getItems() {
        return new ArrayList<>(items); // Retorna uma cópia para evitar mutações externas
    }
}

File: #1.data-structures/#2.Stack/src/kotlin/modules/Stack.kt
class Stack<T> {
  private val items = mutableListOf<T>()

  fun push(item: T) {
      items.add(item)
  }

  fun pop(): T? {
      return if (items.isNotEmpty()) items.removeAt(items.size - 1) else null
  }

  fun peek(): T? {
      return items.lastOrNull()
  }

  fun isEmpty(): Boolean {
      return items.isEmpty()
  }

  fun size(): Int {
      return items.size
  }

  fun clear() {
      items.clear()
  }

  fun getItems(): List<T> {
      return items.toList() // Retorna uma cópia imutável
  }
}
